test/suites/container_devices_nic_bridged.sh

```diff
@@ -429,6 +432,81 @@ test_container_devices_nic_bridged() {
         lxc exec "${ctName}" -- udhcpc6 -f -i eth0 -n -q -t5 2>&1 | grep 'IPv6 obtained'
   fi
 
+  # Check that dnsmasq will resolve on the lo
+  # We can't guarantee that the container has an ip6 addr so just check the A
+  # record; we only care about access to dnsmasq here, not the record itself.
+  dig -4 +retry=0 +notcp @192.0.2.1 A "${ctName}.${dnsDomain}" | grep "${ctName}.${dnsDomain}.\\+0.\\+IN.\\+A.\\+192.0.2."
+  dig -6 +retry=0 +notcp @2001:db8::1 A "${ctName}.${dnsDomain}" | grep "${ctName}.${dnsDomain}.\\+0.\\+IN.\\+A.\\+192.0.2."
+  dig -4 +retry=0 +tcp @192.0.2.1 A "${ctName}.${dnsDomain}" | grep "${ctName}.${dnsDomain}.\\+0.\\+IN.\\+A.\\+192.0.2."
+  dig -6 +retry=0 +tcp @2001:db8::1 A "${ctName}.${dnsDomain}" | grep "${ctName}.${dnsDomain}.\\+0.\\+IN.\\+A.\\+192.0.2."
+
+  # Check that dnsmasq will resolve from the bridge
+  # testImage doesn't have dig, so we create a netns with eth0 connected to the
```

Comment:
Rather than going through this complexity, can you use `ping` instead to ping a host name which implies successful DNS resolution?

Reply:
Yes; it may even allow me to do away with this netns stuff and just use a `testImage` container.

However... I'm not seeing a way to control which protocol DNS resolution is performed over. Since we're testing firewall rules, it would be nice to ensure that all the rules allow/block as intended, not just the ones for UDP.

`ping` could also fail for reasons apart from failed DNS resolution, but this is a less strong argument than the above.

Reply:
hrm that is all true, ok lets leave as is for now.